classes/class.ilUpdateNotificationJob.php

```diff
@@ -0,0 +1,663 @@
+            $ids = intval($records['entity_amount']);
+            $highest_id = intval($records['highest_id']);
+            $newest_date = intval($records['newest_date']);
+
+            return [
+                'id' => $highest_id,
+                'created' => $newest_date,
+                'amount_of_notifications' => $ids,
+            ];
+        }
+        return [
+            'id' => 0,
+            'created' => 0,
+            'amount_of_notifications' => 0,
+        ];
+    }
+
+    /** Query that returns infos about notifications
+     * @return string mysql query
+     */
+    public function getCurrentNotificationInfoQuery() :string {
+        return "SELECT count(`id`) as `entity_amount`, max(`id`) as `highest_id`, max(`create_date`) as `newest_date` FROM il_adn_notifications WHERE `title` LIKE '%Update Notification%' AND `active` = 1;";
+    }
+
+    /** Indicates whether this url exists or not (404 = does not exist)
+     * @param String $url url to check
+     * @return array with status code (e.g. 200/404) and html content
+     */
+    public function fetchUrl(string $url) :array
+    {
+        $ch = curl_init();
+
+        curl_setopt($ch, CURLOPT_URL, $url);
+
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        curl_setopt($ch, CURLOPT_FRESH_CONNECT, true);
+        curl_setopt($ch, CURLOPT_TIMEOUT, self::CURLOPT_TIMEOUT);
+        $content = curl_exec($ch);
+        $status_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        curl_close($ch);
+
+        return [
+          'status_code' => $status_code,
+          'content' => $content
+        ];
+    }
+
+    /** Checks whether this url exists or not.
+     * @param String $url url to check
+     * @return bool true if exists
+     */
+    public function checkUrl(string $url) :bool
+    {
+        $result = $this->fetchUrl($url);
+
+        if ($result['status_code'] == 404) {
+            return false;
+        } else if ($result['content'] === false or empty($result['content'])) {
+            return false;
+        }
+        else {
+            return true;
+        }
+    }
+
+    /** Returns the newest (stable) major version (e.g. 7.0 or 8.0) as string. Checks for up to +5 major versions ahead and returns the first found.
+     * @return string the newest (stable) major version (e.g. 7.0 or 8.0) as string
+     * @throws Exception
+     */
+    public function getNextMajorVersion() :string
+    {
+        $current_version_numeric = strval(ILIAS_VERSION_NUMERIC);
+        $major = intval(explode('.', $current_version_numeric)[0]);
+        $minor = 0;
+        if ($major <= 0) {
+            throw new \Exception("Major version cannot be 0 or lower!");
+        }
+
+        for ($i = 0; $i <= 5; $i++) {
+            $major = ($major + 1);
+
+            $url = $this->settings->get('update_url', self::DEFAULT_UPDATE_URL) . $major . '.' . $minor;
+
+            if ($this->checkUrl($url)) {
+                return "$major.$minor";
```

Comment:
return value sollte hier nur `$major` sein - der Sprachgebrauch in der Community ist "ILIAS 8" und nicht "ILIAS 8.0" um eine Major-Version zu bezeichnen.